web/src/components/DynamicFieldArray.tsx

```diff
@@ -1,60 +1,42 @@
-import {ErrorMessage, Field, FieldArray, useField} from "formik";
-import Image from "next/image";
-import deleteIcon from "~public/icons/close.svg";
-import addIcon from "~public/icons/plus.svg";
+import {ErrorMessage, FieldArray, useField} from "formik";
+import {v4 as uuid} from 'uuid';
 import {styles} from "../styles/tailwindGlobals";
+import {LinkField} from "./LinkField";
 
 interface LinksFieldArrayProps {
   name: string;
+  errors: unknown;
 }
 
-export const DynamicFieldArray = ({ name }: LinksFieldArrayProps) => {
+export const DynamicFieldArray = ({name, errors}: LinksFieldArrayProps) => {
   const [field] = useField(name);
 
   return (
     <div className="w-80 md:w-8/12 md:-mt-5">
       <FieldArray name={name}>
-        {({ push, remove }) => (
+        {({push, remove}) => (
           <div>
             {field.value.map((link: { url: string }, index: number) => (
-              <div className="grid grid-cols-1 space-y-2 mt-2 md:space-y-1">
-                <div className="flex items-center space-x-2">
-                  <Field className={styles.dynamicField} name={`${name}.${index}.url`} />
-                  {index !== 0 && (
-                    <button
-                      type="button"
-                      className="p-1 text-gray-500 hover:text-gray-800 focus:outline-none focus:text-gray-800"
-                      onClick={() => remove(index)}
-                    >
-                      <Image
-                        src={deleteIcon}
-                        className="w-8 px-1 md:w-14 md:px-3"
-                        alt="Delete"
-                      />
-                    </button>
-                  )}
-                  {index === 0 && (
-                    <button type="button" onClick={() => push({ url: '' })}>
-                      <Image
-                        src={addIcon}
-                        className="h-10 w-10 px-2 md:h-16 md:w-16 md:px-4"
-                        alt="Add"
-                      />
-                    </button>
-                  )}
-                </div>
-                <div>
-                  <ErrorMessage
-                    name={`${name}.${index}.url`}
-                    component="div"
-                    className={styles.errorMsg}
-                  />
-                </div>
-              </div>
+              <LinkField
+                name={name}
+                index={index}
+                push={push}
+                remove={remove}
+                key={uuid()}
```

Comment:
Wouldn't this key value change on every render? The `key` value is expected to be the same during the "lifetime" of the component. It tells React if a component has been "moved" (instead of being deleted) which allows it to correctly preserve it's state.

Given that we can only add and remove elements from the end of the listm providing the index as key (and add an [ignore-next-line](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1) for eslint) is a correct way of handling this.

tl;dr: We can use use the index as `key` because we cannot rearrange the list items.